### ISSUE TITLE
Upcast T[] to iterable<mixed,T>

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#702]
https://getpsalm.org/r/6709becd7f
```php
<?php
  class A {}
  class B {}
  /**
   * @param iterable<mixed,A> $p
   */
  function takesIterableOfA(iterable $p): void {}

  takesIterableOfA([new B]); // should complain

  /**
   * @var iterable<mixed,B> $it
   */
  $it = [new B];
  takesIterableOfA($it); // does complain
```
Here, I'd expect first call to `takesIterableOfA` to be complained about, since `array<mixed,B>` is not `iterable<mixed,A>`
